csrc/ir/nodes.cpp

```diff
@@ -3079,6 +3079,35 @@ TensorDomain::TensorDomain(
   resetDomains();
 }
 
+TensorDomain::TensorDomain(
+    IrBuilderPasskey passkey,
+    std::vector<IterDomain*> root_domain,
+    std::vector<int64_t> stride_order,
+    std::vector<std::optional<bool>> contiguity)
+    : Val(passkey, ValType::TensorDomain, DataType::Null),
+      root_domain_(std::move(root_domain)),
+      leaf_domain_(root_domain_),
+      contiguity_(
+          contiguity.empty() ? getContiguityFilledWith(maybeAllocation(), false)
+                             : std::move(contiguity)),
+      has_reduction_(false) {
+  // setting the proper allocation domain
+  if (!stride_order.empty()) {
+    auto rank = root_domain_.size();
+    NVF_ERROR(
```

Comment:
I wish there were an NVF_CHECK_EQ that automatically prints out the value of LHS and RHS for easy debugging... Could be a feature request to work on when you don't feel productive 😄